MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocalizationPluginActivity.java

```diff
@@ -0,0 +1,135 @@
+package com.mapbox.mapboxandroiddemo.examples.plugins;
+
+import android.os.Bundle;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.plugins.localization.LocalizationPlugin;
+import com.mapbox.mapboxsdk.plugins.localization.MapLocale;
+
+/**
+ * Use the localization plugin to retrieve the device's language and set all map text labels to that language.
+ */
+public class LocalizationPluginActivity extends AppCompatActivity implements OnMapReadyCallback {
+
+  private MapView mapView;
+  private LocalizationPlugin localizationPlugin;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_localization_plugin);
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(final MapboxMap mapboxMap) {
+
+    localizationPlugin = new LocalizationPlugin(mapView, mapboxMap);
+
+    findViewById(R.id.language_one_cardview).setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        localizationPlugin.setMapLanguage(MapLocale.ARABIC);
+      }
+    });
+
+
+    findViewById(R.id.language_two_cardview).setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        localizationPlugin.setMapLanguage(MapLocale.RUSSIAN);
+      }
+    });
+
+
+    findViewById(R.id.language_three_cardview).setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        localizationPlugin.setMapLanguage(MapLocale.SIMPLIFIED_CHINESE);
+      }
+    });
+
+    findViewById(R.id.match_map_to_device_language).setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+
+        Snackbar.make(view, R.string.change_device_language_instruction, Snackbar.LENGTH_LONG).show();
+
+        try {
+          localizationPlugin.matchMapLanguageWithDeviceDefault();
```

Comment:
what is potentially throwing the null point exception here? Isn't this a runtime exception?

Reply:
Fixed to runtime exception for when `mapLocale==null` in the `LocalizationPlugin` class.  I also changed the snackbar message because it doesn't need to show the _actual_ exception message. Now it's

```java
catch (RuntimeException exception) {
          Snackbar.make(view, R.string.try_different_language_instruction, Snackbar.LENGTH_LONG).show();
        }
```
with `try_different_language_instruction` being 

```xml
<string name="try_different_language_instruction">Try setting your phone to a different language. German or French perhaps?</string>

```

👇 

![device-2018-03-07-144549](https://user-images.githubusercontent.com/4394910/37122656-42b06904-2216-11e8-8649-21bebc0ef51c.png)